test/integ/cli/test_cli_manifest_upload.py

```diff
@@ -0,0 +1,140 @@
+        """
+
+        # Given a snapshot file:
+        manifest_file = self.create_manifest_file(temp_dir)
+        manifest_file_name = Path(manifest_file).name
+
+        # Now that we have a manifest file, execute the CLI and upload it to S3
+        # The manifest file name is unique, so it will not collide with prior test runs.
+        s3_bucket = os.environ.get("JOB_ATTACHMENTS_BUCKET")
+        runner = CliRunner()
+        # Temporary, always add cli_manifest until launched.
+        main.add_command(cli_manifest)
+        result = runner.invoke(
+            main,
+            [
+                "manifest",
+                "upload",
+                "--s3-cas-uri",
+                f"s3://{s3_bucket}/DeadlineCloud",
+                manifest_file,
+            ],
+        )
+        assert result.exit_code == 0, f"Non-Zeo exit code, CLI output {result.output}"
+
+        # Then validate the Manifest file is uploaded to S3 by checking the file actually exists.
+        manifest_s3_path = f"/DeadlineCloud/Manifests/{manifest_file_name}"
+        s3_client = boto3.client("s3")
+        s3_client.head_object(Bucket=s3_bucket, Key=manifest_s3_path)
+
+        # Cleanup.
+        s3_client.delete_object(Bucket=s3_bucket, Key=manifest_s3_path)
+
+    def test_manifest_upload_by_farm_queue(self, temp_dir):
+        """
+        Simple test to generate a manifest, and then call the upload CLI to uplaod to S3.
+        This test case uses --farm-id and --queue-id
+        The test verifies the manifest is uploaded by doing a S3 get call.
+        """
+
+        # Given a snapshot file:
+        manifest_file = self.create_manifest_file(temp_dir)
+        manifest_file_name = Path(manifest_file).name
+
+        # Now that we have a manifest file, execute the CLI and upload it to S3
+        # The manifest file name is unique, so it will not collide with prior test runs.
+        s3_bucket = os.environ.get("JOB_ATTACHMENTS_BUCKET", "")
+        runner = CliRunner()
+        # Temporary, always add cli_manifest until launched.
+        main.add_command(cli_manifest)
+        result = runner.invoke(
+            main,
+            [
+                "manifest",
+                "upload",
+                "--farm-id",
+                os.environ.get("FARM_ID", ""),
+                "--queue-id",
+                os.environ.get("QUEUE_ID", ""),
+                manifest_file,
+            ],
+        )
+        assert result.exit_code == 0, f"Non-Zeo exit code, CLI output {result.output}"
+
+        # Then validate the Manifest file is uploaded to S3 by checking the file actually exists.
+        manifest_s3_path = f"DeadlineCloud/Manifests/{manifest_file_name}"
+        s3_client = boto3.client("s3")
+        try:
+            s3_client.head_object(Bucket=s3_bucket, Key=manifest_s3_path)
+        except ClientError:
+            assert 0, f"File not found at {s3_bucket}, {manifest_s3_path}"
```

Comment:
nit: we could also just use `pytest.fail` here to always fail: https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest-fail

Reply:
sure